### ISSUE TITLE
QImage.ScanLine returns Pointer to byte

### ISSUE THREAD

**Issue by dsoronda** [dsoronda/QtSharpDemos#9]
``` csharp
                byte* line = image.ScanLine ( rowCount );
```

**Comment by ddobrev** [comment]
Could you please elaborate what the problem here is? Also, in case you take these examples from Zetcode or somewhere else, it would be useful to also post a link to the original and how the particular piece looks in it.

**Comment by dsoronda** [comment]
I found it online somewhere online. It is an example how to optimize
fetching whole row of image as bytes array instead fetching pixel by pixel.

Example is in class included in my last commit ( picture_grayscaledemo.cs)

**Comment by ddobrev** [comment]
I see, and what's the problem you have?

**Comment by ddobrev** [comment]
@dsoronda I saw the C++ code and I assume your problem is that you don't know how to use the returned pointer. I think you should be able to do the following:

```
byte* line = image.ScanLine ( rowCount );
for ( int jj = 0 ; jj < image.Width ; jj++ )
{
    uint rgbpixel = (uint) line[jj * depth]; // I don't know where 'depth' comes from but I assume you know
    ...
}
```

**Comment by ddobrev** [comment]
@dsoronda I have unfortunately not had the time to try this code myself but I think it should work. Let me know if it doesn't so that I try working further on this issue.

**Comment by dsoronda** [comment]
@ddobrev 
Yeah, I'm trying  to keep as little as possible  of unmanaged code in examples. So ill keep an issue but remove a code for now.

**Comment by ddobrev** [comment]
I fully agree, the end goal is definitely not to have any unsafe code in the API. Some cases are difficult to handle, though, so they'll be fixed one by one over time.
